Content.Shared/Storage/EntitySystems/BinSystem.cs

```diff
@@ -76,6 +86,20 @@ private void OnInteractHand(EntityUid uid, BinComponent component, InteractHandE
         args.Handled = true;
     }
 
+    /// <summary>
+    /// Alt interact acts the same as interacting with your hands normally, but allows fallback interaction if the item
+    /// has priority. E.g. a water cup on a water cooler fills itself on a normal click,
+    /// but you can use alternative interactions to restock the cup bin
+    /// </summary>
+    private void OnAltInteractHand(EntityUid uid, BinComponent component, GetVerbsEvent<AlternativeVerb> args)
+    {
+        if (args.Using != null)
+        {
+            var canReach = args.CanAccess && args.CanInteract;
+            OnAfterInteractUsing(uid, component, new AfterInteractUsingEvent(args.User, (EntityUid) args.Using, args.Target, EntityCoordinates.Invalid, canReach));
```

Comment:
you should really just make a private function with this code and call it in both. constructing a new event here is really strange.